feat(GiniHealthSDK): Use payment_state instead of checkIfDocumentIsPayable

razvancapra opened the pull request:
https://ginis.atlassian.net/browse/IPC-296

IPC-296
+ fix tests

motokotomax commented on HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift:
```diff
@@ -20,7 +25,11 @@ import Foundation
-        
+
+    public var isPayable: Bool {
+        extractions.first(where: {$0.name == "payment_state"})?.value == PaymentState.payable.rawValue
```
Proposition: Maybe we can define enum with extraction names?

razvancapra replied:
Thanks for the proposition. Adopted!

motokotomax commented on HealthSDK/GiniHealthSDKExample/GiniHealthSDKExample/AppCoordinator.swift:
```diff
@@ -186,35 +185,34 @@ final class AppCoordinator: Coordinator {
-                                        case .failure(let error):
-                                            print("❌ Checking if document is payable failed: \(String(describing: error))")
+                                        case let .failure(error):
+                                            Log("Obtaining extractions from document with id \(compositeDocument.id) failed with error: \(String(describing: error))", event: .error)
```
should we use Log here? old code use print

zladzeyka reviewed (changes requested):
Great job @razvancapra 
We need to keep the public interface especially method `healthSDK.checkIfDocumentIsPayable` and change the functionality inside this method, otherwise it's a breaking change 🙈

razvancapra commented:
> Great job @razvancapra We need to keep the public interface especially method `healthSDK.checkIfDocumentIsPayable` and change the functionality inside this method, otherwise it's a breaking change 🙈

Reverted and updated!

zladzeyka commented on HealthSDK/GiniHealthSDK/Tests/GiniHealthSDKTests/GiniHealthTests.swift:
```diff
@@ -55,6 +55,34 @@ final class GiniHealthTests: XCTestCase {
         XCTAssertEqual(receivedProviders, expectedProviders)
     }
 
+    func testDcumentIsPayable() {
```
```suggestion
    func testDocumentIsPayable() {
```

zladzeyka commented on HealthAPILibrary/GiniHealthAPILibrary/Sources/GiniHealthAPILibrary/Documents/ExtractionResult.swift:
```diff
@@ -20,7 +39,11 @@ import Foundation
     public var lineItems: [[Extraction]]?
-        
+
+    public var isPayable: Bool {
```
@razvancapra what is the benefit of having 2 interface points that are doing the same thing?
why do we need this variable?

razvancapra replied:
I've added this one so the client doesn't need to handle parsing on this item. But I can remove it and leave it totally on their side.

zladzeyka reviewed (changes requested):
@razvancapra thank you 🙏
Please find a tiny comment from my side.
